library/src/utils/getChecks/getChecks.ts

```diff
@@ -0,0 +1,18 @@
+import type { Pipe, PipeAsync, PipeMeta } from '../../types';
+
+const hasPipeMeta = (val: unknown): val is PipeMeta =>
+  typeof val === `object` && val !== null && `kind` in val && `message` in val;
+
+export const getChecks = (pipe?: Pipe<any> | PipeAsync<any>) =>
```

Comment:
Wouldn’t `unknown` work instead of `any`? As far as I understand, there is a type-guard checking the value 🤔

Reply:
I tried using `unknown` in a few places but ended up getting type errors so I punted on doing extra work. With the API still in flux ahead of a v1, I think it's safe to push those refinements out to a later PR.